fastestimator/op/numpyop/univariate/autocontrast.py

```diff
@@ -0,0 +1,51 @@
+# Copyright 2021 The FastEstimator Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+from typing import Any, Dict, Iterable, List, Union
+
+import numpy as np
+from PIL import Image, ImageOps
+
+from fastestimator.op.numpyop.numpyop import NumpyOp
+from fastestimator.util.traceability_util import traceable
+
+
+@traceable()
+class AutoContrast(NumpyOp):
+    """Adjust image contrast automatically.
+
+    Args:
+        inputs: Key(s) of images to be modified.
+        outputs: Key(s) into which to write the modified images.
+        mode: What mode(s) to execute this Op in. For example, "train", "eval", "test", or "infer". To execute
+            regardless of mode, pass None. To execute in all modes except for a particular one, you can pass an argument
+            like "!infer" or "!train".
+
+    Image types:
+        uint8
+    """
+    def __init__(self,
+                 inputs: Union[str, Iterable[str]],
+                 outputs: Union[str, Iterable[str]],
+                 mode: Union[None, str, Iterable[str]] = None):
+        super().__init__(inputs=inputs, outputs=outputs, mode=mode)
+        self.in_list, self.out_list = True, True
+
+    def forward(self, data: List[np.ndarray], state: Dict[str, Any]) -> List[np.ndarray]:
+        return [self._apply_autocontrast(elem) for elem in data]
+
+    def _apply_autocontrast(self, data: np.ndarray) -> np.ndarray:
+        im = Image.fromarray(data)
+        im = ImageOps.autocontrast(im)
+        return np.copy(np.asarray(im))
```

Comment:
why do we need the np.copy() here?

Reply:
Pytorch throws an error otherwise: The given NumPy array is not writeable, and PyTorch does not support non-writeable tensors. This means you can write to the underlying (supposedly non-writeable) NumPy array using the tensor. You may want to copy the array to protect its data or make it writeable before converting it to a tensor. 

I can use np.array(im) instead of np.copy(np.array(im)) to make it concise and have the same effect.